File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        size_t pos = haystack.find(needle);

        // If find returns std::string::npos, it means the substring was not
        // found.
        if (pos == string::npos) {
            return -1; // Return -1 if the substring is not found
        } else {
            return static_cast<int>(
                pos); // Cast size_t to int and return the position
        }
    }
};

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        hay = len(haystack)
        ned = len(needle)

        if hay < ned:
            return - 1
        for i in range(len(haystack)):
            if haystack[i:i+ned] == needle:
                return i
        return -1